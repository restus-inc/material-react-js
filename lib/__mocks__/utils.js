/* global jest */
const utils = jest.genMockFromModule('../utils');

const newId = () => 'a0123456';
utils.newId = newId;

export default utils;
export { newId };
