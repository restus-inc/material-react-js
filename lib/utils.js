/**
 * Utilities for Material React.
 * @module material-react/lib/utils
 */

/**
 * Generates a rundom ID. The generated ID starts with basic latin child alphabet, and
 * subsequent characters consist of child alphanumeric character. (i.e. it matches
 * `/[a-z][a-z0-9]{digit-1}/`)
 * @param {number} [digit=8] The number of digit of the generated ID
 * @returns {string} The ID generated
 */
export function newId(digit = 8) {
  return (
    (Math.floor(Math.random() * 26) + 10) * (36 ** (digit - 1))
    + Math.floor(Math.random() * (36 ** (digit - 1)))
  ).toString(36);
}

/**
 * Converts the event name to event handler name.
 * @param {string} eventName The event name that is converted to the handler name
 * @return {string} The event handler name converted
 */
export function handlerName(eventName) {
  return `on${eventName.charAt(0).toUpperCase()}${eventName.slice(1)}`;
}
