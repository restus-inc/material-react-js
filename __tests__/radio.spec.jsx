/* global describe, it, expect */
import 'regenerator-runtime/runtime';
import React from 'react';
import { htmlOfRendering } from './utils';

import Radio from '../lib/radio';

describe('Radio component', () => {
  it('supports radio button', () => {
    expect(htmlOfRendering(
      <Radio/>,
    )).resolves.toMatchSnapshot();

    expect(htmlOfRendering(
      <Radio className="baz" disabled={true}/>,
    )).resolves.toMatchSnapshot();

    expect(htmlOfRendering(
      <Radio id="foo" label="bar" defaultChecked={true}/>,
    )).resolves.toMatchSnapshot();

    expect(htmlOfRendering(
      <Radio label="bar" className="baz" name="qux"/>,
    )).resolves.toMatchSnapshot();

    const props = { checked: true };
    function onChange(event) {
      props.checked = event.currentTarget.checked;
    }
    expect(htmlOfRendering(
      <Radio id="foo" label="bar" className="baz" name="qux" checked={props.checked} onChange={onChange}/>,
    )).resolves.toMatchSnapshot();
  });

  it('supports touch target', () => {
    expect(htmlOfRendering(
      <Radio supportsTouch={true}/>,
    )).resolves.toMatchSnapshot();

    expect(htmlOfRendering(
      <Radio className="baz" disabled={true} supportsTouch={true}/>,
    )).resolves.toMatchSnapshot();

    expect(htmlOfRendering(
      <Radio id="foo" label="bar" defaultChecked={true} supportsTouch={true}/>,
    )).resolves.toMatchSnapshot();

    expect(htmlOfRendering(
      <Radio label="bar" className="baz" name="qux" supportsTouch={true}/>,
    )).resolves.toMatchSnapshot();

    const props = { checked: true };
    function onChange(event) {
      props.checked = event.currentTarget.checked;
    }
    expect(htmlOfRendering(
      <Radio id="foo" label="bar" className="baz" name="qux" checked={props.checked} onChange={onChange} supportsTouch={true}/>,
    )).resolves.toMatchSnapshot();
  });
});
