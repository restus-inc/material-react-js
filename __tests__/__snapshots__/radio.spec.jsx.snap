// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component supports radio button 1`] = `
"<div
  class=\\"mdc-radio\\"
>
  <input
    class=\\"mdc-radio__native-control\\"
    type=\\"radio\\"
  />
  <div
    class=\\"mdc-radio__background\\"
  >
    <div
      class=\\"mdc-radio__outer-circle\\"
    />
    <div
      class=\\"mdc-radio__inner-circle\\"
    />
  </div>
  <div
    class=\\"mdc-radio__ripple\\"
  />
</div>"
`;

exports[`Radio component supports radio button 2`] = `
"<div
  class=\\"mdc-radio mdc-radio--disabled baz\\"
>
  <input
    class=\\"mdc-radio__native-control\\"
    disabled=\\"\\"
    type=\\"radio\\"
  />
  <div
    class=\\"mdc-radio__background\\"
  >
    <div
      class=\\"mdc-radio__outer-circle\\"
    />
    <div
      class=\\"mdc-radio__inner-circle\\"
    />
  </div>
  <div
    class=\\"mdc-radio__ripple\\"
  />
</div>"
`;

exports[`Radio component supports radio button 3`] = `
"<div
  class=\\"mdc-form-field\\"
>
  <div
    class=\\"mdc-radio\\"
  >
    <input
      checked=\\"\\"
      class=\\"mdc-radio__native-control\\"
      id=\\"foo\\"
      type=\\"radio\\"
    />
    <div
      class=\\"mdc-radio__background\\"
    >
      <div
        class=\\"mdc-radio__outer-circle\\"
      />
      <div
        class=\\"mdc-radio__inner-circle\\"
      />
    </div>
    <div
      class=\\"mdc-radio__ripple\\"
    />
  </div>
  <label
    for=\\"foo\\"
  >
    bar
  </label>
</div>"
`;

exports[`Radio component supports radio button 4`] = `
"<div
  class=\\"mdc-form-field baz\\"
>
  <div
    class=\\"mdc-radio\\"
  >
    <input
      class=\\"mdc-radio__native-control\\"
      name=\\"qux\\"
      type=\\"radio\\"
    />
    <div
      class=\\"mdc-radio__background\\"
    >
      <div
        class=\\"mdc-radio__outer-circle\\"
      />
      <div
        class=\\"mdc-radio__inner-circle\\"
      />
    </div>
    <div
      class=\\"mdc-radio__ripple\\"
    />
  </div>
  <label>
    bar
  </label>
</div>"
`;

exports[`Radio component supports radio button 5`] = `
"<div
  class=\\"mdc-form-field baz\\"
>
  <div
    class=\\"mdc-radio\\"
  >
    <input
      checked=\\"\\"
      class=\\"mdc-radio__native-control\\"
      id=\\"foo\\"
      name=\\"qux\\"
      type=\\"radio\\"
    />
    <div
      class=\\"mdc-radio__background\\"
    >
      <div
        class=\\"mdc-radio__outer-circle\\"
      />
      <div
        class=\\"mdc-radio__inner-circle\\"
      />
    </div>
    <div
      class=\\"mdc-radio__ripple\\"
    />
  </div>
  <label
    for=\\"foo\\"
  >
    bar
  </label>
</div>"
`;

exports[`Radio component supports touch target 1`] = `
"<div
  class=\\"mdc-touch-target-wrapper\\"
>
  <div
    class=\\"mdc-radio mdc-radio--touch\\"
  >
    <input
      class=\\"mdc-radio__native-control\\"
      type=\\"radio\\"
    />
    <div
      class=\\"mdc-radio__background\\"
    >
      <div
        class=\\"mdc-radio__outer-circle\\"
      />
      <div
        class=\\"mdc-radio__inner-circle\\"
      />
    </div>
    <div
      class=\\"mdc-radio__ripple\\"
    />
  </div>
</div>"
`;

exports[`Radio component supports touch target 2`] = `
"<div
  class=\\"mdc-touch-target-wrapper baz\\"
>
  <div
    class=\\"mdc-radio mdc-radio--disabled mdc-radio--touch\\"
  >
    <input
      class=\\"mdc-radio__native-control\\"
      disabled=\\"\\"
      type=\\"radio\\"
    />
    <div
      class=\\"mdc-radio__background\\"
    >
      <div
        class=\\"mdc-radio__outer-circle\\"
      />
      <div
        class=\\"mdc-radio__inner-circle\\"
      />
    </div>
    <div
      class=\\"mdc-radio__ripple\\"
    />
  </div>
</div>"
`;

exports[`Radio component supports touch target 3`] = `
"<div
  class=\\"mdc-touch-target-wrapper\\"
>
  <div
    class=\\"mdc-form-field\\"
  >
    <div
      class=\\"mdc-radio mdc-radio--touch\\"
    >
      <input
        checked=\\"\\"
        class=\\"mdc-radio__native-control\\"
        id=\\"foo\\"
        type=\\"radio\\"
      />
      <div
        class=\\"mdc-radio__background\\"
      >
        <div
          class=\\"mdc-radio__outer-circle\\"
        />
        <div
          class=\\"mdc-radio__inner-circle\\"
        />
      </div>
      <div
        class=\\"mdc-radio__ripple\\"
      />
    </div>
    <label
      for=\\"foo\\"
    >
      bar
    </label>
  </div>
</div>"
`;

exports[`Radio component supports touch target 4`] = `
"<div
  class=\\"mdc-touch-target-wrapper baz\\"
>
  <div
    class=\\"mdc-form-field\\"
  >
    <div
      class=\\"mdc-radio mdc-radio--touch\\"
    >
      <input
        class=\\"mdc-radio__native-control\\"
        name=\\"qux\\"
        type=\\"radio\\"
      />
      <div
        class=\\"mdc-radio__background\\"
      >
        <div
          class=\\"mdc-radio__outer-circle\\"
        />
        <div
          class=\\"mdc-radio__inner-circle\\"
        />
      </div>
      <div
        class=\\"mdc-radio__ripple\\"
      />
    </div>
    <label>
      bar
    </label>
  </div>
</div>"
`;

exports[`Radio component supports touch target 5`] = `
"<div
  class=\\"mdc-touch-target-wrapper baz\\"
>
  <div
    class=\\"mdc-form-field\\"
  >
    <div
      class=\\"mdc-radio mdc-radio--touch\\"
    >
      <input
        checked=\\"\\"
        class=\\"mdc-radio__native-control\\"
        id=\\"foo\\"
        name=\\"qux\\"
        type=\\"radio\\"
      />
      <div
        class=\\"mdc-radio__background\\"
      >
        <div
          class=\\"mdc-radio__outer-circle\\"
        />
        <div
          class=\\"mdc-radio__inner-circle\\"
        />
      </div>
      <div
        class=\\"mdc-radio__ripple\\"
      />
    </div>
    <label
      for=\\"foo\\"
    >
      bar
    </label>
  </div>
</div>"
`;
